test/path-parsing.spec.js

```diff
@@ -113,4 +113,27 @@ describe('gulp-rename path parsing', function () {
       helper(srcPattern, obj, null, done, { multiExt: true });
     });
   });
+
+  describe('original file context', function () {
+    it('passed to plugin as second argument', function (done) {
+      var srcPattern = 'test/fixtures/hello.min.txt';
+      var obj = function (path, file) {
+        file.should.be.instanceof(Object);
```

Comment:
Can you just have this check for attributes existing and being the right values on the `file` param? These two checks aren't needed.